packages/dashboard/src/customization/widgets/lineSymbol/lineAnchor/component.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { CSSProperties, useEffect, useRef } from 'react';
+import { LineWidget } from '../types';
+import { UpdateWidgetsAction } from '~/store/actions';
+import { useDrag } from 'react-dnd';
+import { ItemTypes } from '~/components/dragLayer/itemTypes';
+import { DashboardWidget } from '~/types';
+import { useGridSettings } from '~/components/actions/useGridSettings';
+
+export const LineAnchor: React.FC<{
+  style: CSSProperties;
+  dotType: 'start' | 'end';
+  widget: LineWidget;
+  updateWidget: (widget: DashboardWidget) => UpdateWidgetsAction;
+}> = ({ style, dotType, widget, updateWidget }) => {
+  const initialPositionRef = useRef({
+    initialStartX: 0,
+    initialStartY: 0,
+    initialEndX: 0,
+    initialEndY: 0,
+  });
+
+  const { cellSize } = useGridSettings();
+
+  const [{ diff, isDragging }, ref] = useDrag({
+    type: ItemTypes.LineAnchor,
+    item: () => {
+      initialPositionRef.current = {
+        initialStartX: widget.properties.startX ?? 0,
+        initialStartY: widget.properties.startY ?? 0,
+        initialEndX: widget.properties.endX ?? 0,
+        initialEndY: widget.properties.endY ?? 0,
+      };
+      return {
+        dotType,
+      };
+    },
+    collect: (monitor) => {
+      const offsetDifference = monitor.getDifferenceFromInitialOffset();
+      return { diff: offsetDifference, isDragging: monitor.isDragging() };
+    },
+  });
+  useEffect(() => {
+    if (isDragging && diff) {
+      let updatedWidget;
+      if (dotType === 'start') {
+        const newStartX = Math.max(
+          0,
+          Math.min((initialPositionRef.current.initialStartX ?? 0) + diff.x, Math.floor(widget.width * cellSize))
+        );
+        const newStartY = Math.max(
+          0,
+          Math.min((initialPositionRef.current.initialStartY ?? 0) + diff.y, Math.floor(widget.height * cellSize))
+        );
+        updatedWidget = {
+          ...widget,
+          properties: {
+            ...widget.properties,
+            startX: newStartX,
+            startY: newStartY,
+          },
+        };
+      } else if (dotType == 'end') {
+        const newEndX = Math.max(
+          0,
+          Math.min((initialPositionRef.current.initialEndX ?? 0) + diff.x, Math.floor(widget.width * cellSize))
```

Comment:
sorry, your question seems a bit ambiguous. 

are you suggesting i use `Math.round` instead of `Math.floor`? minor difference, but i think that's a good idea - will implement.

or are you asking whether it already is an integer? yeah, the start/end coordinates are always whole numbers because diff only has integers and `Math.round` (previously `Math.floor`) returns integers.